### PR TITLE
Add forecasting by regression

### PR DESCRIPTION
**Pull request by LittleLittleCloud**

### CODE REVIEW

File: csharp-scenarios/AutoML.Net/Sonar-Forecasting.ipynb [hunk 1]
```diff
@@ -0,0 +1,507 @@
+{
+  "cells": [
```
Comment (JakeRadMSFT) [inline csharp-scenarios/AutoML.Net/Sonar-Forecasting.ipynb:2]: @LittleLittleCloud Can you put this under - 

machine-learning/

and the dataset under 

machine-learning/data

**Comment by JakeRadMSFT** [comment]
Data under 'machine-learning/data' please

**Comment by JakeRadMSFT** [comment]
> Data under 'machine-learning/data' please

Then it will match what Luis does in his PR.

File: machine-learning/Luna-Forecasting.ipynb [hunk 2]
```diff
@@ -0,0 +1,507 @@
...
+            ]
+          },
+          "metadata": {},
+          "output_type": "display_data"
+        }
+      ],
+      "source": [
+        "// call df.Head(10) if you want to see the first 10 line.\n",
```
Comment (luisquintanilla) [inline machine-learning/Luna-Forecasting.ipynb:243]: Move this comment into a markdown

File: machine-learning/Luna-Forecasting.ipynb [hunk 3]
```diff
@@ -0,0 +1,507 @@
...
+        "df.Tail(10)"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {},
+      "source": [
+        "### Construct training pipeline\n",
+        "The following code shows how to construct pipeline, which simply concatenate all columns starts with \"prev\" as a single-vector feature and feed it into regressors.\n",
```
Comment (luisquintanilla) [inline machine-learning/Luna-Forecasting.ipynb:252]:
```suggestion
        "The following code shows how to construct an ML.NET training pipeline which concatenates all the feature columns with the `prev` prefix into a single feature vector and provides it as input to the regression trainer",
```

File: machine-learning/Luna-Forecasting.ipynb [hunk 4]
```diff
@@ -0,0 +1,507 @@
...
+        "\n",
+        "Notice that the pipeline calls `mlContext.Auto().Regression` in the end, which returns all available regressors with pre-defined search space. Then we can pass this pipeline into `AutoMLExperiemnt` for hpo.\n"
```
Comment (luisquintanilla) [inline machine-learning/Luna-Forecasting.ipynb:254]:
```suggestion
        "Notice that the pipeline calls `mlContext.Auto().Regression` in the end, which returns all available regressors with pre-defined search spaces. Then we can pass this pipeline into `AutoMLExperiment` for hyperparameter optimization.\n"
```

File: machine-learning/Luna-Forecasting.ipynb [hunk 5]
```diff
@@ -0,0 +1,507 @@
...
+        "dotnet_interactive": {
+          "language": "csharp"
+        },
+        "vscode": {
+          "languageId": "dotnet-interactive.csharp"
+        }
+      },
+      "outputs": [],
+      "source": [
+        "var rowCount = df.Rows.Count();\n",
```
Comment (luisquintanilla) [inline machine-learning/Luna-Forecasting.ipynb:270]: I'd add more details on what you're doing here. Maybe even splitting up the code into the individual steps.

File: machine-learning/Luna-Forecasting.ipynb [hunk 6]
```diff
@@ -0,0 +1,507 @@
...
+            ]
+          },
+          "metadata": {},
+          "output_type": "display_data"
+        }
+      ],
+      "source": [
+        "// Configure AutoML\n",
```
Comment (luisquintanilla) [inline machine-learning/Luna-Forecasting.ipynb:363]: I would convert the inline code commets to Markdown cells and also split up the code into individual steps and add more details:

- Split train/test data
- Initialize monitor
- Create experiment

Comment (luisquintanilla) [reply machine-learning/Luna-Forecasting.ipynb:363]: Also, is there a reason we call `Result` directly on the async operation. We should use

`var res = await experiment.Run()`;

File: machine-learning/Luna-Forecasting.ipynb [hunk 7]
```diff
@@ -0,0 +1,507 @@
...
+            "text/html": [
+              "<table><thead><tr><th>MeanAbsoluteError</th><th>MeanSquaredError</th><th>RootMeanSquaredError</th><th>LossFunction</th><th>RSquared</th></tr></thead><tbody><tr><td><div class=\"dni-plaintext\">116358.10044642857</div></td><td><div class=\"dni-plaintext\">27160304042.780903</div></td><td><div class=\"dni-plaintext\">164803.83503663045</div></td><td><div class=\"dni-plaintext\">27160304098.598053</div></td><td><div class=\"dni-plaintext\">0.9783342123594955</div></td></tr></tbody></table>"
+            ]
+          },
+          "metadata": {},
+          "output_type": "display_data"
+        }
+      ],
+      "source": [
+        "var model = res.Model;\n",
```
Comment (luisquintanilla) [inline machine-learning/Luna-Forecasting.ipynb:413]: Add a markdown section here describing what this code does.

**Review by luisquintanilla — CHANGES_REQUESTED** [review]
Looking good @LittleLittleCloud . I added some comments.

File: machine-learning/Luna-Forecasting.ipynb [hunk 8]
```diff
@@ -0,0 +1,507 @@
...
+              "    \r\n",
+              "</div    \r\n"
+            ]
+          },
+          "metadata": {},
+          "output_type": "display_data"
+        }
+      ],
+      "source": [
+        "var lastThreeWeek = df[\"load\"].Cast<float>().TakeLast(7 * 24 * 3);\n",
```
[the remark this answers is not in the document]
Comment (LittleLittleCloud) [reply machine-learning/Luna-Forecasting.ipynb]: It's plotting three weeks data and we are using previous 2 week as features

File: machine-learning/Luna-Forecasting.ipynb (hunk 6 above)
Comment (LittleLittleCloud) [reply machine-learning/Luna-Forecasting.ipynb]: Would that makes cell too small which causes a notebook having too many cells to run? A long cell might frighten users off, but too many cells might also use up one's patience..

**Review by JakeRadMSFT — APPROVED** [review]

**Pull request merged**